### PR TITLE
Add Additional PI Status Codes

### PR DESCRIPTION
**Pull request by zurcher**
## Description

Add definitions for PI Status Codes from PI Spec 1.8A, Volume 3: Shared Architectural Elements, Chapter 6.6.

- [x] Impacts functionality?
- [ ] Impacts security?
- [x] Breaking change?
  Changes namespace for status code value definitions
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Built an external project using some items defined in this module.

## Integration Instructions

N/A

### CODE REVIEW

**Comment by zurcher** [comment]
See "Commits" tab for a side-by-side comparison to PiStatusCode.h

**Review by joschock — APPROVED** [review]

**Review by vineelko — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -35,4 +35,5 @@ pub mod dxe_services;
 pub mod fw_fs;
 pub mod hob;
 pub mod list_entry;
+pub mod pi_status_code;
```
Comment (makubacki) [inline src/lib.rs:38]: I suggest just using `status_code`.

File: src/pi_status_code.rs
Comment (makubacki) [inline src/pi_status_code.rs:1]: These should be reconciled with what's already in https://github.com/microsoft/mu_rust_pi/blob/initial_changes/src/protocols/status_code.rs which is a subset of status codes that are either used or likely to be used.

File: src/pi_status_code.rs [hunk 2]
```diff
@@ -0,0 +1,1104 @@
...
+
+//
+// Required for IA32, X64, IPF, ARM and EBC defines for CPU exception types
+//
+use r_efi::efi::{Guid, protocols::debug_support};
+
+///
+/// Status Code Type Definition.
+///
+pub type EfiStatusCodeType = u32;
```
Comment (makubacki) [inline src/pi_status_code.rs:25]: Was there a reason for the `Type` suffix?

Comment (zurcher) [reply src/pi_status_code.rs:25]: The spec defines Code Types and Code Values as separate data types

Comment (makubacki) [reply src/pi_status_code.rs:25]: Makes sense. I got caught up on the Type being related to this being a type without thinking about its context in status codes.

File: src/pi_status_code.rs
Comment (zurcher) [reply src/pi_status_code.rs:1]: Moved all status codes to lib/status_code.rs

**Comment by zurcher** [comment]
Rust doesn't like importing two modules named status_code :/

**Comment by makubacki** [comment]
> Rust doesn't like importing two modules named status_code :/

For the benefit of others, as we discussed offline, `mu_pi::protocols::status_code` and `mu_pi::status_code` are fine.

Consumers should not try to map both to `status_code`. Something like `use mu_pi::{protocol, status_code}` can allow access to both like `protocol::status_code` or `status_code::<whatever>`.

We've generally been trying to include the relevant part of the namespace like `protocol::` to help make it clear in usage that a protocol is being used.

**Pull request merged**